### PR TITLE
Controller Input

### PR DESCRIPTION
**Pull request by cha2251**
* Can now take input from an Xbox controller
* Can now play chords
* By default, A is mapped to a note, and X is a chord

### CODE REVIEW

File: src/button_input/controller.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        self.add_controller_note = add_controller_note
+        self.active = True
+        self.controller_listener = threading.Thread(target=self.listener, args=())
+        self.controller_listener.daemon = True
+        self.controller_listener.start()
+
+    def listener(self):
+        while self.active:
+            print('GOING')
```
Comment (DavidTheFighter) [inline src/button_input/controller.py:61]: Probably should remove this lol

File: test/button_input_tests/button_input_test.py [hunk 2]
```diff
@@ -3,21 +3,26 @@
 from pynput import keyboard
```
Comment (DavidTheFighter) [inline test/button_input_tests/button_input_test.py:3]: Fix the unit tests timing out on the CI!

File: src/button_input/controller.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            'BTN_THUMBR': ControllerButton.RightThumb,
+            'BTN_SELECT': ControllerButton.Back,
+            'BTN_START': ControllerButton.Start,
+            'BTN_DPAD_LEFT': ControllerButton.LeftDPad,
+            'BTN_DPAD_RIGHT': ControllerButton.RightDPad,
+            'BTN_DPAD_UP': ControllerButton.UpDPad,
+            'BTN_DPAD_DOWN': ControllerButton.DownDPad
+        }
+
+        self.add_controller_note = add_controller_note
```
Comment (DavidTheFighter) [inline src/button_input/controller.py:53]: I don't like the name of this callback much, it doesn't convey immediately that this is a callback. Just a nitpick tho.

File: requirements.txt [hunk 4]
```diff
@@ -1,37 +1,48 @@
-PyQt5
-matplotlib
-numpy
 altgraph==0.17.3
 attrs==22.1.0
 beautifulsoup4==4.11.1
 certifi==2022.6.15
 charset-normalizer==2.1.1
 colorama==0.4.5
+contourpy==1.0.7
```
Comment (DavidTheFighter) [inline requirements.txt:7]: Where are these extra dependencies coming from? Do we need these?

**Review by DavidTheFighter — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 4 above)
Comment (cha2251) [reply requirements.txt:7]: This is all coming from actually specifying a version of matplotlib. De we still need that?

**Review by DavidTheFighter — APPROVED** [review]
LGTM

**Pull request merged**